automation/test.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2017 Red Hat, Inc.
+#
+
+# CI considerations: $TARGET is used by the jenkins build, to distinguish what to test
+# Currently considered $TARGET values:
+#     kubernetes-release: Runs all functional tests on a release kubernetes setup
+#     openshift-release: Runs all functional tests on a release openshift setup
+
+set -ex
+
+export WORKSPACE="${WORKSPACE:-$PWD}"
+readonly ARTIFACTS_PATH="exported-artifacts"
+
+mkdir -p "$ARTIFACTS_PATH"
+
+if [[ $TARGET =~ openshift-.* ]]; then
+  export KUBEVIRT_PROVIDER="os-3.10.0"
+elif [[ $TARGET =~ .*-1.10.4-.* ]]; then
+  export KUBEVIRT_PROVIDER="k8s-1.10.4"
+fi
+
+export KUBEVIRT_NUM_NODES=2
+
+kubectl() { cluster/kubectl.sh "$@"; }
+
+export CDI_NAMESPACE="${CDI_NAMESPACE:-kube-system}"
+
+make cluster-down
```

Comment:
should we need to do the down?  If the answer is "to remove stale containers from a previous run" then what happens if the previous run used a different KUBEVIRT_PROVIDER?  In that case the cluster-down would not find those old containers at all.

Reply:
This is more for when you are running the whole testing suite locally. It does no harm in the CI, but if I manually run test.sh it helps to tear down the previous cluster I had.

Reply:
>  if the previous run used a different KUBEVIRT_PROVIDER

I thought the cluster-up/cluster-down stuff is namespaced by provider now. I know at one point the goal was to be able to run openshift and k8s clusters in parallel on the same machine. I *think* that's possible now.

so, make cluster-down should only remove stale containers from a previous run for the same provider.

Reply:
I am just using the cluster-down part for when I am running through an entire run on my local machine. I don't think it has any effect in the CI environment.